app/models/mail_log.rb

```diff
@@ -31,10 +31,23 @@ class MailLog < ActiveRecord::Base
 
   validates :mail_hash, uniqueness: { case_sensitive: false }
 
-  def self.build(mail)
-    mail_log = new
-    mail_log.mail = mail
-    mail_log
+  scope :list, -> { order(updated_at: :desc) }
+
+  ### CLASS METHODS
+
+  class << self
+    def build(mail)
+      mail_log = new
+      mail_log.mail = mail
+      mail_log
+    end
+
+    def in_year(year)
```

Comment:
Könnten man auch vereinfachen, auch in `Event` mittels `Date.new(year.presence || Time.zone.now.year, 1,1).all_year`